src/model/channel/attachment_type.rs

```diff
@@ -0,0 +1,84 @@
+use std::borrow::Cow;
+#[cfg(not(feature = "tokio"))]
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+#[cfg(feature = "tokio")]
+use tokio::fs::File;
```

Comment:
This has caused a compilation error in the [multipart code of the HTTP client](https://github.com/serenity-rs/serenity/runs/4662315938?check_suite_focus=true#step:8:143). Personally, this should always be `tokio`'s `File`. I don't see a point branching the types here, considering `tokio` will always be available if the `http` (or `gateway`) feature is enabled, and the HTTP client is the only place where `AttachmentType` is used.

Reply:
This actually made me wonder, why has it been moved to `serenity::model::channel`? How is it related to channels? Does it have to be moved from the `serenity::http` module?

Reply:
`AttachmentType` is a member in `CreateInteractionResponseFollowup`. 
https://github.com/serenity-rs/serenity/blob/4f7dad7c96b0b2111502325090147beacc04b5ce/src/builder/create_interaction_response_followup.rs#L11-L14
The `builder` feature don't select `http` or `gateway`, if the file is included as `File` in `AttachmentType`, it will have to be forked, unless `tokio` is added to `builder` which wont make sense.
Or we can use something else for `CreateInteractionResponseFollowup`

Reply:
> `AttachmentType` is used in various places in `builder` as a member. 

I apologise, I should've used a better word. When I said "...and the HTTP client is the only place where `AttachmentType` is used." I meant it's the sole location where `AttachmentType` is used to alter behaviour, i.e. *where it is consumed*. If you disabled the `http` feature, it would lose its purpose.

That is why I don't agree moving `AttachmentType` to `serenity::model::channel`. It should stay where it is.

> The `builder` feature don't select `http` or `gateway`, if the file is included as `File` in `AttachmentType`, it will have to be forked, unless `tokio` is added to `builder` - which wont make sense.

True. Ideally, `builder` wouldn't have to depend on either of those, but there is no other elegant way of defining attachments in a message other than `AttachmentType`, and I'd like to avoid introducing another type for attachments just for builders. As a compromise, the builders that use `AttachmentType` should be feature-gated behind `http`. Subsequently, we wouldn't have to choose between `std::fs::File` or `tokio::fs::File` depending on whether `tokio` is present or not.

Reply:
The reason for this PR is to make the lib usable with other request implementations. In such case, the models without the `model` feature and request builders are the only components used. Therefore `AttachmentType` is needed without the `http` feature selected.